### ISSUE TITLE
If simple trigger is created long before scheduled, the first fire time is not computed well

### ISSUE THREAD

**Issue by ismdiego** [quartznet/quartznet#2455]
**Describe the bug**

If you create a simple trigger long before is later scheduled, the first fire time is computed as the StartTimeUtc, which is not correct. This causes problems with repetitions (if any) because the nextFireTimeUtc is also not calculated correctly.

**Version used**

I have tested this with v3.10.0

**To Reproduce**

You can see a working example here:
https://github.com/ismdiego/quartz_simple_trigger_reuse

There is one job defined, that delays for 1 second.

There are two triggers for the same job:
- trigger1 (repeats forever, each 10 seconds)
- trigger2 (repeats 3 times, each 2 seconds) -> scheduled after trigger1 completes

You can see that the second repetition of the trigger2 starts before it must.

**Expected behavior**

- trigger1 fires at 00:00
- trigger1 completes at 00:01
- trigger2 fires, first repetition at 00:01
- trigger2 second repetition at 00:03 -> error, it fires at 00:02 (2 seconds after 00:00)
- trigger2 third repetition at 00:05 -> error, it fires at 00:04 (2 seconds after 00:02)
- trigger1 fires at 00:10
- repeat indefinitely... the next repetitions work well

Maybe this is not the best practice (to create a trigger and a long time afterwards schedule it) but I think the system should behave as expected.

**Additional context**

I have programmed a quick fix and it works for my case. It also passes all the Unit tests defined in Quartz.Tests.Unit. I attach the git patch that can be applied over the v3.10.0 tag:
[0001-Fixes-2455.patch](https://github.com/user-attachments/files/16109402/0001-Fixes-2455.patch)

**Comment by lahma** [comment]
Without going too deep into logic in question, wouldn't it be easier to dynamically create the trigger on demand when needed (start time would reflect the actual request)?

**Comment by ismdiego** [comment]
Yes, I agree. In fact, this is what I have learnt from my tests: do not create and store the triggers, but instead create them just before their use. In my design, this implies storing some trigger factories instead of the actual triggers.

But I think that this is probably a bug and it's something that can be improved, and so I tried with those small changes in code. With those changes you can use both approaches and it seems to be safe (all unit tests pass).

I did not prepare a PR following the "[Being a good open source citizen](https://hackernoon.com/being-a-good-open-source-citizen-9060d0ab9732#.x3hocgw85)" practices as you recommend, but I can do it if needed. I would like to help only if this does not bother you or your current roadmap (v4.x).

**Comment by lahma** [comment]
I think PR against main (v4) with accompanying test coverage would be appreciated.

**Comment by lahma** [comment]
Fixed in #2958 (3.x) and ported to main in #2972.
